nexus/tests/integration_tests/unauthorized_coverage.rs

```diff
@@ -0,0 +1,141 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use super::endpoints::*;
+use expectorate::assert_contents;
+use openapiv3::OpenAPI;
+use std::collections::BTreeMap;
+
+/// Checks for uncovered public API endpoints
+///
+/// This test compares the endpoints covered by the "unauthorized" test with
+/// what's in the OpenAPI spec for the public API to make sure all endpoints are
+/// accounted-for.
+#[test]
+fn test_unauthorized_coverage() {
+    // Load the OpenAPI schema for Nexus's public API.
+    let schema_path = "../openapi/nexus.json";
+    let schema_contents = std::fs::read_to_string(&schema_path)
+        .expect("failed to read Nexus OpenAPI spec");
+    let spec: OpenAPI = serde_json::from_str(&schema_contents)
+        .expect("Nexus OpenAPI spec was not valid OpenAPI");
+
+    // Take each operation that we find in the OpenAPI spec, make a regular
+    // expression that we can use to match against it (more on this below), and
+    // throw them into a BTreeMap.
+    let mut spec_operations: BTreeMap<Operation, regex::Regex> = spec
+        .operations()
+        .map(|(path, method, op)| {
+            // We're going to take URLs from our test cases and match them
+            // against operations in the API spec.  The URLs from the API spec
+            // contain variables (e.g., "/instances/{instance_name}").  Our test
+            // cases have those variables filled in already (e.g.,
+            // "/instances/my-instance").
+            //
+            // To match a URL from the test case against one from the OpenAPI
+            // spec, we're going to:
+            //
+            // - use a regular expression to replace `{varname}` in the API
+            //   spec's URL with `[^/]+` (one or more non-slash characters)
+            //
+            // - use that string as the basis for a second regex that we'll
+            //   store with the Operation.  We'll use this second regex to match
+            //   URLs to their operation.
+            //
+            // This is slow (lookups will take time linear in the total number
+            // of API endpoints) and a little cheesy, but it's expedient and
+            // robust enough for our purposes.
+            //
+            // This will fail badly if it turns out that the URL contains any
+            // characters that would be interpreted specially by the regular
+            // expression engine.  So let's check up front that those aren't
+            // present.
+            assert!(
+                path.chars().all(|c| c.is_ascii_alphanumeric()
+                    || c == '_'
+                    || c == '-'
+                    || c == '{'
+                    || c == '}'
+                    || c == '/'),
+                "unexpected character in URL: {:?}",
+                path
+            );
+            let re = regex::Regex::new("/\\{[^.}][^}]+\\}").unwrap();
```

Comment:
I was trying to avoid matching wildcard path variables, since I don't think this would work for that case.  Your question caused me dig into the dropshot code and I realized it's really `{foo:.*}`, not `{.*}` as I'd thought, so I'll just remove this leading `.`.  It won't do the right thing for wildcard variables, but it will fail explicitly when it finds them.  We may want to update this to handle the wildcard case better when we add tests for those.   But the only such endpoint that we have right now is both unpublished and accessible even to unauthenticated users (console assets).